subsys/bluetooth/host/conn.c

```diff
@@ -562,6 +562,10 @@ static bool send_frag(struct bt_conn *conn, struct net_buf *buf, uint8_t flags,
 	k_sem_give(bt_conn_get_pkts(conn));
 	if (tx) {
 		conn_tx_destroy(conn, tx);
+		/* `buf` might not get destroyed, and its `tx` pointer will still be reachable.
+		 * Make sure that we don't try to use the destroyed context later.
+		 */
+		tx_data(buf)->tx = NULL;
```

Comment:
This PR kind of does this part in 3 different ways. 

The first one here we store `tx_data(buf)->tx` as a local variable and use that for the NULL check and the call for `conn_tx_destroy`, and then set `tx_data(buf)->tx = NULL;`. 

The next one is arguable simpler and does 
```c
		if (tx_data(buf)->tx) {
			conn_tx_destroy(conn, tx_data(buf)->tx);
			tx_data(buf)->tx = NULL;
		}
```
and is very easy to follow. 

Then you have the third one where you first store it as a local variable `struct bt_conn_tx *tx = tx_data(buf)->tx;`, then set `tx_data(buf)->tx = NULL;`, then do the NULL check on the local variable and then finally call `conn_tx_destroy(conn, tx);`

Might be worth to follow the same pattern in all 3 for the sake of readability.

Reply:
very much agree. I originally had (and preferred) number 2 (the one with the code block) but @alwa-nordic wanted me to use num 3 instead.

I would've preferred the second one because it's simpler to read, but `send_frag` reads better with the `tx` local, so I'l use the third method everywhere.